packages/itwin/tree-widget/src/components/trees/VisibilityTreeEventHandler.ts

```diff
@@ -129,7 +139,7 @@ export class VisibilityTreeEventHandler extends UnifiedSelectionTreeEventHandler
       .pipe(
         mergeMap((changes) => this.changeVisibility(changes)),
       )
-      .subscribe();
+      .subscribe({ complete: () => { this._isChangingVisibility = false; }, error: () => { this._isChangingVisibility = false; } });
```

Comment:
I think after `_isChangingVisibility` flag is reset to false, `updateCheckboxes` should be called to make sure that they are updated after all the event that were possibly skipped.